cmd/optimize/report.go

```diff
@@ -123,11 +123,46 @@ func listReports(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	reportRows, err := extractReportData(resp)
+	mainDataSet := resp.Main()
+	if mainDataSet == nil {
+		output.PrintCmdStatus(cmd, "No results found for given input\n")
+		return nil
+	}
+
+	reportRows, err := extractReportData(mainDataSet)
 	if err != nil {
 		return fmt.Errorf("extractReportData: %w", err)
 	}
 
+	_, next_ok := mainDataSet.Links["next"]
```

Comment:
Should we instead add a new method in uql (or an optional `follow` parameter to `uql.ExecuteQuery`) that will follow the links and return the full results? 

This is a common functionality that would be needed in multiple uses of the uql package, including the `fsoc uql` command.

Reply:
Let's keep this implementation to fix the optimize report and also provide an example. An internal request has been open for the `uql` package (UQL-4595)